"Name already registered with serializer" with parcel@next

<!---
Thanks for filing an issue 😄 ! Before you submit, please read the following:

Search open/closed issues before submitting since someone might have asked the same thing before!
-->

# 🐛 bug report

When I run `parcel build` I receive a "Name already registered with serializer" error
<!--- Provide a general summary of the issue here -->

## 🎛 Configuration (.babelrc, package.json, cli command)

Blank project with yarn2#[pnp](https://next.yarnpkg.com/features/pnp) and parcel@next

## 🤔 Expected Behavior

No error while building

## 😯 Current Behavior

```
/yarnparcel/.yarn/cache/@parcel-core-npm-2.0.0-nightly.65-a3573223f8-1.zip/node_modules/@parcel/core/lib/serializer.js:28
    throw new Error('Name already registered with serializer');
    ^

Error: Name already registered with serializer
    at registerSerializableClass (/yarnparcel/.yarn/cache/@parcel-core-npm-2.0.0-nightly.65-a3573223f8-1.zip/node_modules/@parcel/core/lib/serializer.js:28:11)
    at Object.<anonymous> (/yarnparcel/.yarn/$$virtual/@parcel-package-manager-virtual-dab9d78bc4/0/cache/@parcel-package-manager-npm-2.0.0-nightly.67-6d59c584b3-1.zip/node_modules/@parcel/package-manager/lib/Npm.js:82:37)
    at Module._compile (internal/modules/cjs/loader.js:956:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:973:10)
    at Module.load (internal/modules/cjs/loader.js:812:32)
    at Function.module_1.Module._load (/yarnparcel/.pnp.js:24084:14)
    at Module.require (internal/modules/cjs/loader.js:849:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.<anonymous> (/yarnparcel/.yarn/$$virtual/@parcel-package-manager-virtual-dab9d78bc4/0/cache/@parcel-package-manager-npm-2.0.0-nightly.67-6d59c584b3-1.zip/node_modules/@parcel/package-manager/lib/index.js:19:12)
    at Module._compile (internal/modules/cjs/loader.js:956:30)
```

## 💁 Possible Solution

No Idea, sorry ¯\\_(ツ)_/¯

## 🔦 Context

I'm trying to setup a  project with bleeding-edge tooling :)

## 💻 Code Sample

This installs yarn@berry *GLOBALLY*, creates a sample project and produces the error

```
mkdir yarnparcel
cd yarnparcel
npm i -g yarn@berry
yarn init
yarn add parcel@next
echo "<h1>Hello</h1>" > index.html
yarn run parcel build index.html
```
## 🌍 Your Environment

<!--- Include as many relevant details about the environment you experienced the bug in -->

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 2.0.0-alpha.3.2
| Node             | 12.13.0
| Yarn         | 2.0.0-rc.27
| Operating System | MacOS 10.14

<!-- Love parcel? Please consider supporting our collective:
👉  https://opencollective.com/parcel/donate -->

I have the same error regarding version `2.0.0-nightly.714`
But at the version `2.0.0-beta.3.1` build is ok, but I still need the `2.0.0-nightly.714` version to try to work around the floating bug https://github.com/parcel-bundler/parcel/issues/4145 and some other floating bugs like: 

```bash
[14:16:24] :	 [Step 3/5] Error: Command failed: parcel build templates/partials/header.pug               --dist-dir ./build/zfx --public-url /wp-content/themes/zfx/assets --no-scope-hoist --no-source-maps
[14:16:24] :	 [Step 3/5]     at checkExecSyncError (child_process.js:640:11)
[14:16:24] :	 [Step 3/5]     at execSync (child_process.js:676:15)
[14:16:24] :	 [Step 3/5]     at Object.<anonymous> (/src/build-docker.js:12:1)
[14:16:24] :	 [Step 3/5]     at Module._compile (internal/modules/cjs/loader.js:1068:30)
[14:16:24] :	 [Step 3/5]     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1097:10)
[14:16:24] :	 [Step 3/5]     at Module.load (internal/modules/cjs/loader.js:933:32)
[14:16:24] :	 [Step 3/5]     at Function.Module._load (internal/modules/cjs/loader.js:774:14)
[14:16:24] :	 [Step 3/5]     at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)
[14:16:24] :	 [Step 3/5]     at internal/main/run_main_module.js:17:47 {
[14:16:24] :	 [Step 3/5]   status: null,
[14:16:24] :	 [Step 3/5]   signal: 'SIGSEGV',
[14:16:24] :	 [Step 3/5]   output: [ null, null, null ],
[14:16:24] :	 [Step 3/5]   pid: 23,
[14:16:24] :	 [Step 3/5]   stdout: null,
[14:16:24] :	 [Step 3/5]   stderr: null
[14:16:24] :	 [Step 3/5] }
[14:16:24] :	 [Step 3/5]  [0m [91mnpm ERR! code ELIFECYCLE
[14:16:24] :	 [Step 3/5]  [0m [91mnpm  [0m [91mERR! errno [0m [91m 1
[14:16:24] :	 [Step 3/5]  [0m [91mnpm ERR! main-site@1.16.0 build-docker: `node build-docker.js`
[14:16:24] :	 [Step 3/5] npm  [0m [91mERR! Exit status 1
[14:16:24] :	 [Step 3/5] npm [0m [91m ERR! 
[14:16:24] :	 [Step 3/5]  [0m [91mnpm ERR! Failed at the main-site@1.16.0 build-docker script.
[14:16:24] :	 [Step 3/5]  [0m [91mnpm ERR! This is probably not a problem with npm. There is likely additional logging output above.
[14:16:24] :	 [Step 3/5]  [0m [91m
[14:16:24] :	 [Step 3/5]  [0m [91mnpm ERR! A complete log of this run can be found in:
[14:16:24] :	 [Step 3/5] npm ERR!     /root/.npm/_logs/2021-06-09T13_16_24_204Z-debug.log
```

NOPE.
Although my previous advice on expanding docker memory works to fix unexpected build interruption, but when I try to install any other version than `2.0.0-beta.3.1` I get the same error:
```bash
MacBook-Pro-Andrey:client andrew$ npm run build-wp

> main-site@1.16.0 build-wp /Users/andrew/Sites/main-site/src/client
> node build-wp.js

Error: Name already registered with serializer
    at registerSerializableClass (/Users/andrew/Sites/main-site/src/client/node_modules/@parcel/core/lib/serializer.js:30:11)
    at Object.<anonymous> (/Users/andrew/Sites/main-site/src/client/node_modules/@parcel/core/node_modules/@parcel/fs/lib/NodeFS.js:229:39)
    at Module._compile (/Users/andrew/Sites/main-site/src/client/node_modules/v8-compile-cache/v8-compile-cache.js:192:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (/Users/andrew/Sites/main-site/src/client/node_modules/v8-compile-cache/v8-compile-cache.js:159:20)
    at Object.<anonymous> (/Users/andrew/Sites/main-site/src/client/node_modules/@parcel/core/node_modules/@parcel/fs/lib/index.js:21:15)
    at Module._compile (/Users/andrew/Sites/main-site/src/client/node_modules/v8-compile-cache/v8-compile-cache.js:192:30)
child_process.js:655
    throw err;
    ^

Error: Command failed: parcel build templates/partials/header.pug               --dist-dir ../wordpress/wp-content/themes/zfx/assets/ --public-url /wp-content/themes/zfx/assets --no-optimize
    at checkExecSyncError (child_process.js:616:11)
    at execSync (child_process.js:652:15)
    at Object.<anonymous> (/Users/andrew/Sites/main-site/src/client/build-wp.js:12:1)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)
    at internal/main/run_main_module.js:17:47 {
  status: 1,
  signal: null,
  output: [ null, null, null ],
  pid: 7875,
  stdout: null,
  stderr: null
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! main-site@1.16.0 build-wp: `node build-wp.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the main-site@1.16.0 build-wp script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/andrew/.npm/_logs/2021-06-15T07_24_38_333Z-debug.log
MacBook-Pro-Andrey:client andrew$
```

--- a/packages/core/core/src/serializer.js
+++ b/packages/core/core/src/serializer.js
@@ -11,10 +11,6 @@ const nameToCtor: Map<string, Class<*>> = new Map();
 const ctorToName: Map<Class<*>, string> = new Map();
 
 export function registerSerializableClass(name: string, ctor: Class<*>) {
-  if (nameToCtor.has(name)) {
-    throw new Error('Name already registered with serializer');
-  }
-
   if (ctorToName.has(ctor)) {
     throw new Error('Class already registered with serializer');
   }

